python/ray/train/xgboost/xgboost_trainer.py

```diff
@@ -156,17 +211,58 @@ def __init__(
                 'Upgrade with: `pip install -U "xgboost>=1.7"`'
             )
 
-        # TODO(justinvyu): [Deprecated] Remove in 2.11
-        if dmatrix_params != _DEPRECATED_VALUE:
-            raise DeprecationWarning(
-                "`dmatrix_params` is deprecated, since XGBoostTrainer no longer "
-                "depends on the `xgboost_ray.RayDMatrix` utility. "
-                "You can remove this argument and use `dataset_config` instead "
-                "to customize Ray Dataset ingestion."
+        # TODO(justinvyu): [Deprecated] Legacy XGBoostTrainer API
+        legacy_api = train_loop_per_worker is None
+        if legacy_api:
+            train_loop_per_worker = self._get_legacy_train_fn_per_worker(
+                xgboost_train_kwargs=train_kwargs,
+                run_config=run_config,
+                label_column=label_column,
+                num_boost_round=num_boost_round,
+                datasets=datasets,
             )
+            train_loop_config = params or {}
+
+        super(XGBoostTrainer, self).__init__(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config=train_loop_config,
+            xgboost_config=xgboost_config,
+            scaling_config=scaling_config,
+            run_config=run_config,
+            datasets=datasets,
+            dataset_config=dataset_config,
+            resume_from_checkpoint=resume_from_checkpoint,
+            metadata=metadata,
+        )
+
+    def _get_legacy_train_fn_per_worker(
+        self,
+        xgboost_train_kwargs: Dict,
+        run_config: Optional[ray.train.RunConfig],
+        datasets: Optional[Dict[str, GenDataset]],
+        label_column: Optional[str],
+        num_boost_round: Optional[int],
+    ) -> Callable[[Dict], None]:
+        """Get the training function for the legacy XGBoostTrainer API."""
+
+        if not datasets or not datasets.get(TRAIN_DATASET_KEY):
+            raise ValueError(
+                "`datasets` must be provided for the legacy XGBoostTrainer API. "
+                "This dict must contain the training dataset under the "
+                f"key '{TRAIN_DATASET_KEY}'."
+            )
+        if not label_column:
+            raise ValueError(
+                "`label_column` must be provided for the legacy XGBoostTrainer API. "
+                "This is the column name of the label in the dataset."
+            )
+
+        num_boost_round = num_boost_round or 10
+
+        _log_deprecation_warning(LEGACY_XGBOOST_TRAINER_DEPRECATION_MESSAGE)
```

Comment:
Let's hold off on explicitly logging that it's deprecated until we have the GH issue and documentation for V2 published?

Reply:
sounds good, will comment out for now